Build fails on Linux 5.6.8-zen1 due to incompatible pointer type

I believe the issue is caused by upstream commit https://github.com/torvalds/linux/commit/c3c6201ad35b8ca05ff782161d87f639a83de5a9, which changed the type of `rate_control_ops.alloc` struct member.

```
DKMS make.log for r8822be-21 for kernel 5.6.8-zen1-1-zen (x86_64)
Thu Apr 30 10:07:41 CEST 2020
make: Entering directory '/usr/lib/modules/5.6.8-zen1-1-zen/build'
  AR      /var/lib/dkms/r8822be/21/build/built-in.a
  CC [M]  /var/lib/dkms/r8822be/21/build/cam.o
  CC [M]  /var/lib/dkms/r8822be/21/build/base.o
  CC [M]  /var/lib/dkms/r8822be/21/build/debug.o
  CC [M]  /var/lib/dkms/r8822be/21/build/core.o
  CC [M]  /var/lib/dkms/r8822be/21/build/efuse.o
  CC [M]  /var/lib/dkms/r8822be/21/build/ps.o
  CC [M]  /var/lib/dkms/r8822be/21/build/rc.o
  CC [M]  /var/lib/dkms/r8822be/21/build/regd.o
/var/lib/dkms/r8822be/21/build/rc.c:288:11: error: initialization of ‘void * (*)(struct ieee80211_hw *)’ from incompatible pointer type ‘void * (*)(struct ieee80211_hw *, struct dentry *)’ [-Werror=incompatible-pointer-types]
  288 |  .alloc = rtl_rate_alloc,
      |           ^~~~~~~~~~~~~~
/var/lib/dkms/r8822be/21/build/rc.c:288:11: note: (near initialization for ‘rtl_rate_ops.alloc’)
  CC [M]  /var/lib/dkms/r8822be/21/build/stats.o
cc1: some warnings being treated as errors
make[1]: *** [scripts/Makefile.build:268: /var/lib/dkms/r8822be/21/build/rc.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [Makefile:1683: /var/lib/dkms/r8822be/21/build] Error 2
make: Leaving directory '/usr/lib/modules/5.6.8-zen1-1-zen/build'
```

--- a/rc.c
+++ b/rc.c
@@ -15,6 +15,7 @@
 #include "wifi.h"
 #include "base.h"
 #include "rc.h"
+#include <linux/version.h>
 
 /*
  *Finds the highest rate index we can use
@@ -250,7 +251,13 @@ static void rtl_rate_update(void *ppriv,
 {
 }
 
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(5, 4, 35) || \
+	(LINUX_VERSION_CODE >= KERNEL_VERSION(5, 5, 0) && \
+	LINUX_VERSION_CODE <= KERNEL_VERSION(5, 6, 7))
 static void *rtl_rate_alloc(struct ieee80211_hw *hw, struct dentry *debugfsdir)
+#else
+static void *rtl_rate_alloc(struct ieee80211_hw *hw)
+#endif
 {
 	struct rtl_priv *rtlpriv = rtl_priv(hw);
 	return rtlpriv;

